### PR TITLE
Added synchronization with the main storage device before direct recording

### PR DESCRIPTION
**Pull request by Roman-Koshelev**
Added reset of caching stream buffers before direct writing to the descriptor

### CODE REVIEW

**Review by vitaut — CHANGES_REQUESTED** [review]
Thanks for the PR but please revert unrelated formatting changes and add more context. I applied clang-format separately.

**Comment by Roman-Koshelev** [comment]
Done.

File: include/fmt/ostream.h [hunk 1]
```diff
@@ -38,24 +38,24 @@ auto get_file(std::filebuf&) -> FILE*;
...
     c_file = fbuf->file();
   else
     return false;
-  if (c_file) return write_console(c_file, data);
-#else
-  ignore_unused(os, data);
 #endif
-  return false;
+  if (!c_file) return false;
+  os.flush();
```
Comment (vitaut) [inline include/fmt/ostream.h:55]: We should only flush the buffer when writing to a console. I suggest moving the ostream change to a separate PR and addressing this issue. The rest LGTM.

**Review by vitaut — APPROVED** [review]

**Comment by vitaut** [comment]
Merged without the ostream part in https://github.com/fmtlib/fmt/commit/3b7f58a8b3e35cfee57558a41e3e897b0ba7106a. Thanks.

File: include/fmt/format-inl.h [hunk 2]
```diff
@@ -1433,13 +1432,12 @@ extern "C" __declspec(dllimport) int __stdcall WriteConsoleW(  //
     void*, const void*, dword, dword*, void*);
 
 FMT_FUNC bool write_console(std::FILE* f, string_view text) {
-  auto fd = _fileno(f);
+  int fd = _fileno(f);
   if (!_isatty(fd)) return false;
+  std::fflush(f);
```
Comment (dimztimz) [inline include/fmt/format-inl.h:1437]: Adding `fflush` here is not correct. This function is called either from the C stream print() or the C++ iostream print(). In the latter case, fflush should not be called at all, instread the ostream flush should be invoked.

Comment (vitaut) [reply include/fmt/format-inl.h:1437]: This flush is indeed redundant for iostream but in the future please report issues like these once to avoid too much notification noise.
